scrapers_next/in/people.py

```diff
@@ -12,48 +16,60 @@ def process_page(self):
             title = titles[0].text_content()
             p.extras["title"] = title
 
-        assistant = (
-            CSS("div .fusion-text.fusion-text-2 p").match(self.root)[0].text_content()
-        )
-        assistant = re.search(r"Legislative Assistant:\s(.+)", assistant).groups()[0]
-
-        phones = (
-            CSS("div .fusion-text.fusion-text-2 p").match(self.root)[1].text_content()
-        )
-        phone1, phone2 = re.search(
-            r"Phone:\s(\d{3}-\d{3}-\d{4})\s\|\s(.+)", phones
-        ).groups()
+        info_div = CSS("div .fusion-text.fusion-text-2 p").match(self.root)
+        for field in info_div:
+            content = field.text_content().strip()
+            if ":" in content:
+                label, detail = [x.strip() for x in content.split(":")]
+                if label == "Legislative Assistant":
+                    p.extras[label] = detail
+                elif label == "Phone":
+                    phones = re.findall(r"\d{3}-\d{3}-\d{4}", detail)
+                    if phones:
+                        p.capitol_office.voice = phones[0]
+                    extra_num = 1
+                    for extra_phone in phones[1:]:
+                        p.extras[f"Additional Phone {extra_num}"] = extra_phone
+                elif label == "Email":
+                    p.email = label
+                elif label == "Media Contact":
+                    media_contact = [x.strip() for x in detail.split("|")]
+                    p.extras["Media Contact Name"] = media_contact[0]
+                    p.extras["Media Contact Email"] = media_contact[1]
+                else:
+                    raise NewDetailFieldEncountered
 
-        media_contact = (
-            CSS("div .fusion-text.fusion-text-2 p").match(self.root)[3].text_content()
-        )
-        media_contact_name, media_contact_email = re.search(
-            r"Media Contact:\s(\w+\s\w+)\s\|\s(.+)", media_contact
-        ).groups()
         addr = (
             XPath("//div/div[3]/div/div[2]/div/div[1]/text()")
             .match(self.root)[0]
             .strip()
         )
 
-        twitter = CSS("div .fusion-social-links a").match(self.root)[0].get("href")
-        twitter_id = re.search(r"https://twitter\.com/(.+)", twitter).groups()[0]
-
-        fb = CSS("div .fusion-social-links a").match(self.root)[1].get("href")
-        fb_id = (
-            re.search(r"https://(www\.)?facebook\.com/(.+)", fb).groups()[1].rstrip("/")
-        )
+        social_dict = {
+            "facebook": None,
+            "instagram": None,
+            "twitter": None,
+            "youtube": None,
+        }
+        social_links = CSS("div .fusion-social-links a").match(self.root)
+        for link in social_links:
+            href = link.get("href").lower()
+            for soc in social_dict.keys():
+                if soc in href:
+                    raw_handle = re.search(rf"(.+)({soc}\.com/)(.+)", href).groups()[-1]
```

Comment:
Where possible, we should pre-compile regexes. It's good practice and is _slightly_ faster. Particularly in loops.

Reply:
Great catch! My newest commit now precompiles the regex patterns for both [phone](https://github.com/openstates/openstates-scrapers/pull/4233/commits/8160a417548f9091286faa87c41610a294b0c520#diff-9eae9c00f2832ba4d89c6294f928772d6d554299758b67c70db01777d9283412R20-R28) and [social media](https://github.com/openstates/openstates-scrapers/pull/4233/commits/8160a417548f9091286faa87c41610a294b0c520#diff-9eae9c00f2832ba4d89c6294f928772d6d554299758b67c70db01777d9283412R20-R28).